epp-client/src/domain/transfer.rs

```diff
@@ -4,466 +4,90 @@ use epp_client_macros::*;
-        (self.request, self.extension)
-    }
-}
-
-#[derive(Debug)]
-pub struct DomainTransferQuery<E> {
-    request: DomainTransferReq,
-    extension: Option<E>,
-}
-
-impl<E: EppExtension> EppRequest<E> for DomainTransferQuery<E> {
-    type Input = DomainTransferReq;
-    type Output = DomainTransferResponse;
-
-    fn into_parts(self) -> (Self::Input, Option<E>) {
-        (self.request, self.extension)
-    }
-}
-
-/// Type that represents the &lt;epp&gt; request for transfer request for domain
-///
-/// ## Usage
-///
-/// ```no_run
-/// use std::collections::HashMap;
-///
-/// use epp_client::config::{EppClientConfig, RegistryConfig};
-/// use epp_client::EppClient;
-/// use epp_client::domain::transfer::DomainTransferRequest;
-/// use epp_client::common::NoExtension;
-/// use epp_client::login::Login;
-/// use epp_client::logout::Logout;
-///
-/// #[tokio::main]
-/// async fn main() {
-///     // Create a config
-///     let mut registry: HashMap<String, RegistryConfig> = HashMap::new();
-///     registry.insert(
-///         "registry_name".to_owned(),
-///         RegistryConfig {
-///             host: "example.com".to_owned(),
-///             port: 700,
-///             tls_files: None,
-///         },
-///     );
-///     let config = EppClientConfig { registry };
-///
-///     // Create an instance of EppClient, passing the config and the registry you want to connect to
-///     let mut client = match EppClient::new(&config, "registry_name").await {
-///         Ok(client) => client,
-///         Err(e) => panic!("Failed to create EppClient: {}",  e)
-///     };
-///
-///     let login = Login::<NoExtension>::new("username", "password", None);
-///     client.transact(login, "transaction-id").await.unwrap();
-///
-///     // Create an DomainTransferRequest instance
-///     let domain_transfer_request = DomainTransferRequest::<NoExtension>::new(
-///         "eppdev-100.net", None, "epP4uthd#v"
-///     );
-///
-///     // send it to the registry and receive a response of type DomainTransferRequestResponse
-///     let response = client.transact(domain_transfer_request, "transaction-id").await.unwrap();
-///
-///     println!("{:?}", response);
-///
-///     let logout = Logout::<NoExtension>::new();
-///     client.transact(logout, "transaction-id").await.unwrap();
-/// }
-/// ```
-impl<E: EppExtension> DomainTransferRequest<E> {
-    pub fn new(
-        name: &str,
-        years: Option<u16>,
-        auth_password: &str,
-    ) -> DomainTransferRequest<NoExtension> {
-        DomainTransferRequest {
-            request: DomainTransferReq {
-                operation: "request".to_string(),
-                domain: DomainTransferReqData {
-                    xmlns: XMLNS.to_string(),
-                    name: name.into(),
-                    period: years.map(Period::new),
-                    auth_info: Some(DomainAuthInfo::new(auth_password)),
-                },
+impl DomainTransferRequest {
```

Comment:
How about instead of having `DomainTransferRequest` and `DomainTransferUpdate` we have something like `DomainTransfer` with methods query, request, approve, reject, cancel?

Reply:
Because of the different `Transaction` impls I think we'll still want there to be different types. Where we put the methods is sort of a separate question, they could also be free-standing functions in `domain::transfer` I guess?

Reply:
Resolved by rebasing on top of #43.